src/redismodule.h

```diff
@@ -830,6 +834,7 @@ typedef struct RedisModuleKeyOptCtx RedisModuleKeyOptCtx;
 typedef int (*RedisModuleCmdFunc)(RedisModuleCtx *ctx, RedisModuleString **argv, int argc);
 typedef void (*RedisModuleDisconnectFunc)(RedisModuleCtx *ctx, RedisModuleBlockedClient *bc);
 typedef int (*RedisModuleNotificationFunc)(RedisModuleCtx *ctx, int type, const char *event, RedisModuleString *key);
+typedef void (*RedisModulePostExecUnitJobFunc) (RedisModuleCtx *ctx, void *pd);
```

Comment:
@yossigo one example could be the unlink event we're adding in #9406

Reply:
@MeirShpilraien we discussed this in a core-team meeting.
i'm not 100 on board with that, but the decision was to narrow the scope of this API (be only about KSN), so that maybe we can extend it int he future, rather than make it wide now (a generic "execution unit"), and later regret it.
this applies just to the API. we can keep the internal concept of execution units, where we're free to keep adjusting it, just not expose it to users via API.

Reply:
@oranagra I have updated the PR and the top comment accordingly. External API specify only key space notifications.